extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java

```diff
@@ -55,72 +56,73 @@ public void startVaultContainers(BuildProducer<DevServicesConfigResultBuildItem>
 
         // figure out if we need to shut down and restart any existing Vault container
         // if not and the Vault container have already started we just return
-        if (closeables != null) {
+        if (closeable != null) {
             boolean restartRequired = !currentDevServicesConfiguration.equals(capturedDevServicesConfiguration);
             if (!restartRequired) {
                 return;
             }
-            for (Closeable closeable : closeables) {
-                try {
-                    closeable.close();
-                } catch (Throwable e) {
-                    log.error("Failed to stop Vault container", e);
-                }
+            try {
+                closeable.close();
```

Comment:
you can't be sure that by the time you reach the `close`, you still have the same variable, that it has not been closed by somebody else, and/or nulled-out. I don't believe `volatile` gives you and guarantee.
if it is `volatile`, I suppose it may be accessed from multiple threads? or you know for a fact that between the `if` and the `close` nobody can touch the variable?

Reply:
This is for starting dev services, I don't believe parallel called to start of the same extension is possible.

Reply:
this seems reasonnable.
I can see the same pattern in the `DevServicesDatasourceProcessor`. so you are probably right.